src/FunFair.CodeAnalysis.Tests/ProhibitedMethodsDiagnosticsAnalyzerTests.cs

```diff
@@ -265,6 +265,77 @@ void Test()
             this.VerifyCSharpDiagnostic(test, expected);
         }
 
+        [Fact]
+        public void QueryArbitrarySqlAsyncIsBanned()
+        {
+            const string test = @"
+    using System;
+
+    namespace FunFair.Common.Data
+    {
+         public interface ISqlServerDatabase
+         {
+                Task QueryArbitrarySqlAsync(string sql);
+         }
+    }
+
+    namespace ConsoleApplication1
+    {
+        class TypeName
+        {
+            void Test(FunFair.Common.Data.ISqlServerDatabase sqlServerDatabase, string sql)
+            {
+                sqlServerDatabase.QueryArbitrarySqlAsync(sql)
+            }
+        }
+    }";
+            DiagnosticResult expected = new DiagnosticResult
+                                        {
+                                            Id = "FFS0007",
+                                            Message = @"Only use ISqlServerDatabase.QueryArbitrarySqlAsync in integration tests",
+                                            Severity = DiagnosticSeverity.Error,
+                                            Locations = new[] {new DiagnosticResultLocation(path: "Test0.cs", line: 18, column: 17)}
+                                        };
+
+            this.VerifyCSharpDiagnostic(test, expected);
+        }
+
+        [Fact]
+        public void ExecuteArbitrarySqlAsyncIsBanned()
+        {
+            const string test = @"
+    using System;
+
+    namespace FunFair.Common.Data
+    {
+         public interface ISqlServerDatabase
+         {
+                Task QueryArbitrarySqlAsync(string sql);
```

Comment:
Should this be ExecuteArbitrarySqlAsync?

Reply:
Test is not failing due this error. It is event fine if interface member is commented like this
`namespace FunFair.Common.Data
    {
         public interface ISqlServerDatabase
         {
                //Task QueryArbitrarySqlAsync(string sql);
         }
    }`
I will change it to ExecuteArbitrarySqlAsync to avoid any confusion in future.